### ISSUE TITLE
Site Editor: Options menu: improve clarity and accessibility of the panel switchers

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#57552]
### Description

Part of a broader review of the Options menu.

The Options menu shows a couple additional items only in the Site editor. These items are sort of 'panel switchers' that allow users to switch the panel from the Settings one to the Styles one. Screenshot:

![Screenshot 2024-01-04 at 16 17 23](https://github.com/WordPress/gutenberg/assets/1682452/62e4e3c3-400e-4f4f-881f-775c00bff24f)

**Logical grouping and clarity**
As its name hints to, the 'Plugins' group in the Options menu is meant to show items added by plugins. The Settings snd Styles panel swichers are arguably 'plugins'. As a user, I'd be pretty confused about what a 'plugin' is in this context.
Okacing these two items within the Plugins group isn't ideal froma logical perspective and doesn't help clarity.
Ideally, they should be moved to their own group or into an existing group that better identifies their purpose.

When there is no support for Global styles, the 'Styles' item isn't shown, which makes sense. However, having the 'Plugins' group that only shows a 'Settings' item is even more confusing. Screenshot:

![Screenshot 2024-01-04 at 16 45 23](https://github.com/WordPress/gutenberg/assets/1682452/79e849fb-d51a-4c88-aacd-dbf3f29d99f1)

As a user, I'd wonder what this 'Settings plugin' is and I'd be pretty confused.

**Redundancy**
To my understanding, these two items have been added to provide users with the ability to switch the panel when the corresponding buttons in the top bar are not available (e.g. on small screens). However, on large screens they just repeat the same functionality that is already available in the top bar. See screenshot.
Ideally, these two items should be shown only when needed.

**Consistency**
I'm not sure why equivalent items are not available iin the _Post editor_ options menu. To me, the Options menu should be as much consistent as possible. 

**Accessibility**
Under the hood, these two items render two button elements with a `role="menuitemcheckbox"`. That's incorrect as only one of the two items can be selected at a time. They should use a `role="menuitemradio"`, that is no different from what is already done for the 'Visual editor' and 'Code editor' items. More specifically:
- When there's more than one item, they should use `role="menuitemradio"`,
- When there is only one item, for example with no items added by plugins and no support for global styles, the 'Settings' item should use `role="menuitemcheckbox"` or maybe be entirely removed.

### Step-by-step reproduction instructions

- Go to the Site editor and open the Options menu.
- Observe the 'Plugins' group within the Options menu.
- Try to establish wheter there is a logical relationship between the 'Settings' / 'Styles' items and the group name 'Plugins'.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
